contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y

```diff
@@ -1640,33 +1643,34 @@ tsql_pivot_expr: TSQL_PIVOT '(' func_application FOR columnref IN_P in_expr ')'
 					a_star_restarget->indirection = NIL;
 					a_star_restarget->val = (Node *) a_star;
 					a_star_restarget->location = -1;
-					
 
 					/* prepare aggregation function for pivot source sql */
 					restarget_aggfunc->name = NULL;
 					restarget_aggfunc->name_location = -1;
 					restarget_aggfunc->indirection = NIL;
 					restarget_aggfunc->val = (Node *) $3;
 					restarget_aggfunc->location = -1;
-					
-					if (IsA((List *)$7, List))
+
+					column_list = (List *)$8;
+					column_const_list = NIL;
+					value_col = list_nth_node(ColumnRef, ((FuncCall *) $3)->args, 0);
+					value_colstr = list_nth_node(String, value_col->fields, ((List *)value_col->fields)->length - 1);
```

Comment:
Thanks for the review, I'll add a function args check in gram-tsql-rule.y

Reply:
Addressed this issue in grammar rule definition. 
In TSQL PIVOT definition, the aggregation function only support  1 column argument, so I changed the rule to the following one:
```
tsql_pivot_expr: TSQL_PIVOT '(' func_name '(' columnref ')' FOR columnref IN_P '(' columnList ')' ')'
```
This will restrict the aggregate function argument to be not null and must be a column ref and there could be only 1 argument in the aggregate function. We don't need to explicitly check it here because bison parser will check it for us.